File: python/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/whole_part/Membresia.py
from datetime import date
from .DatosCliente import DatosCliente
from ..ProductoCine import ProductoCine

class Membresia(ProductoCine):
    
    datosCliente = DatosCliente()
    fechaVencimiento : date
    
    def gestionarProductoCine(self):
        print("hacer gestion de membresia")

File: php/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/membresias/Membresia.php
<?php

/**
 * Description of Membresia
 *
 * @author giltenorio
 */
class Membresia implements ProductoCine{
    public $datosCliente;
    public $dia;
    public $mes;
    public $anio;
    
    public function __construct($datosCliente, $dia, $mes, $anio) {
        $this->datosCliente = $datosCliente;
        $this->dia = $dia;
        $this->mes = $mes;
        $this->anio = $anio;
    }
    public function gestionarProductoCine() {
        echo 'Gestionando producto Cine Membresia';
    }
    
    public function mostrarMembresia(){
        return $this->datosCliente->getDatosCliente()."Fecha de vencimiento: ". $this->dia."/". $this->mes."/". $this->anio;
    }

   
}
